assets/src/domain/shared/entities/datetimes/helpers/status.ts

```diff
@@ -0,0 +1,30 @@
+import { Datetime } from '@edtrServices/apollo/types';
+import isActive from '../predicates/isActive';
+import isExpired from '../predicates/isExpired';
+import isSoldOut from '../predicates/isSoldOut';
+import isTrashed from '../../../services/predicates/isTrashed';
+import isUpcoming from '../predicates/isUpcoming';
+
+import { DATETIME_STATUS_ID } from '../constants';
+
+const status = (date: Datetime): string => {
+	if (isActive(date)) {
+		return DATETIME_STATUS_ID.ACTIVE;
+	}
+	if (isExpired(date)) {
+		return DATETIME_STATUS_ID.EXPIRED;
+	}
+	if (isSoldOut(date)) {
+		return DATETIME_STATUS_ID.SOLD_OUT;
+	}
+	if (isTrashed(date)) {
+		return DATETIME_STATUS_ID.TRASHED;
+	}
+	if (isUpcoming(date)) {
+		return DATETIME_STATUS_ID.UPCOMING;
+	}
+
+	return DATETIME_STATUS_ID.INACTIVE;
+};
```

Comment:
IMHO, the order of priority (highest first) should be:
- `TRASHED`
- `EXPIRED`
- `SOLD_OUT`
- `UPCOMING`
- `ACTIVE`
- `INACTIVE` - fallback

Reply:
Thanks, updated.